app/models/case_log.rb

```diff
@@ -34,15 +34,21 @@ class CaseLog < ApplicationRecord
   belongs_to :managing_organisation, class_name: "Organisation"
 
   scope :for_organisation, ->(org) { where(owning_organisation: org).or(where(managing_organisation: org)) }
-  scope :filter_by_status, ->(status) { where status: status }
-  scope :filter_by_years, lambda { |years|
+  scope :filter_by_status, ->(status, user = nil) { where status: status }
+  scope :filter_by_years, lambda { |years, user = nil|
     first_year = years.shift
     query = filter_by_year(first_year)
     years.each { |year| query = query.or(filter_by_year(year)) }
     query.all
   }
   scope :filter_by_year, ->(year) { where(startdate: Time.utc(year.to_i, 4, 1)...Time.utc(year.to_i + 1, 4, 1)) }
 
+  scope :filter_by_user, lambda { |selected_user, user|
+                           if !selected_user.include?("all") && user.present?
```

Comment:
then this could just be:
```
where(id: PaperTrail::Version.where(item_type: "CaseLog", event: "create", whodunnit: user.to_global_id.uri.to_s).map(&:item_id)) unless user == "all"
```